provider-example/src/hash.rs

```diff
@@ -0,0 +1,42 @@
+use rustls::crypto::hash;
+use sha2::Digest;
+
+pub struct SHA256;
```

Comment:
nit: in `hmac.rs` you're using `Sha256Hmac` and here `SHA256`, maybe match case with the upstream `sha2` crate and prefer camelcase vs uppercase?